src/service/resources-impl.js

```diff
@@ -624,6 +624,39 @@ export class Resources {
   }
 
   /**
+   * Requires the layout of the specified element or top-level sub-elements
+   * within.
+   * @param {!Element} element
+   * @param {number=} opt_parentPriority
+   * @return {!Promise}
+   * @restricted
+   */
+  requireLayout(element, opt_parentPriority) {
+    const promises = [];
+    this.discoverResourcesForElement_(element, resource => {
+      if (resource.getState() == ResourceState.LAYOUT_COMPLETE) {
+        return;
+      }
+      if (resource.getState() != ResourceState.LAYOUT_SCHEDULED) {
+        resource.measure();
+        if (resource.isDisplayed()) {
+          resource.element.whenBuilt().then(() => {
+            this.scheduleLayoutOrPreload_(
+                resource,
+                /* layout */ true,
+                opt_parentPriority,
+                /* forceOutsideViewport */ true);
+          });
+        }
+      }
+      if (resource.isDisplayed()) {
+        promises.push(resource.loadedOnce());
```

Comment:
Good catch. Made me realize it was a slight bug - the first `isDisplayed`/`measure` should have been inside the promise. Fixed now.